### PR TITLE
add kb about ssc and circleci

### PR DESCRIPTION
**Pull request by sebasrevuelta**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [ ] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Review by khorne3 — APPROVED** [review]
Thank you! I edited the intro a bit, but otherwise all looks good

**Comment by armchairlinguist** [comment]
I made some edits here as well - @sebasrevuelta or @khorne3 let me know if you have any concerns about my changes. Mostly they're adding links and context. I also deleted the stray copy of the yaml that crept in somehow.

File: docs/kb/semgrep-supply-chain/ssc-lockfiles-circleci.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+In CircleCI, you can generate a lockfile during the first job and then pass it to the Semgrep scan using a [workspace](https://circleci.com/docs/workspaces/) to share files between jobs.
+
+The following `config.yml` file demonstrates how you can generate a lockfile and pass it to subsequent jobs using CircleCI workspaces. This example uses a `maven_dep_tree.txt` file, which [typically needs to be generated](/docs/semgrep-supply-chain/setup-maven/) from a `pom.xml` for Maven dependency tracking.
+
+```yaml
+version: 2.1
+
+jobs:
+  build:
```
Comment (armchairlinguist) [inline docs/kb/semgrep-supply-chain/ssc-lockfiles-circleci.md:21]: Maybe we should give this a more specific name than `build`?

**Review by armchairlinguist — APPROVED** [review]

File: docs/kb/semgrep-supply-chain/ssc-lockfiles-circleci.md (hunk 1 above)
Comment (sebasrevuelta) [reply docs/kb/semgrep-supply-chain/ssc-lockfiles-circleci.md]: yes, I agree. I changed that to lock_file_generation. Thanks @armchairlinguist and @khorne3!!

**Pull request merged**